web/pay-ui/app/components/common/DateRangeFilter.vue

```diff
@@ -0,0 +1,256 @@
+  },
+  {
+    code: DATEFILTER_CODES.YESTERDAY,
+    label: t(`enum.DATEFILTER_CODES.${DATEFILTER_CODES.YESTERDAY}`),
+    getRange: getYesterday
+  },
+  {
+    code: DATEFILTER_CODES.LASTWEEK,
+    label: t(`enum.DATEFILTER_CODES.${DATEFILTER_CODES.LASTWEEK}`),
+    getRange: getLastWeek
+  },
+  {
+    code: DATEFILTER_CODES.LASTMONTH,
+    label: t(`enum.DATEFILTER_CODES.${DATEFILTER_CODES.LASTMONTH}`),
+    getRange: getLastMonth
+  },
+  {
+    code: DATEFILTER_CODES.CUSTOMRANGE,
+    label: t(`enum.DATEFILTER_CODES.${DATEFILTER_CODES.CUSTOMRANGE}`),
+    getRange: resetRange
+  }
+]
+
+function luxonToCalendarDate(date: DateTime): CalendarDate {
+  return new CalendarDate(date.year, date.month, date.day)
+}
+
+function getToday() {
+  const today = new Date()
+  const date = new CalendarDate(today.getFullYear(), today.getMonth() + 1, today.getDate())
+  return { start: date, end: date }
+}
+
+function getYesterday() {
+  const yesterday = new Date()
+  yesterday.setDate(yesterday.getDate() - 1)
+  const date = new CalendarDate(yesterday.getFullYear(), yesterday.getMonth() + 1, yesterday.getDate())
+  return { start: date, end: date }
+}
+
+function getLastWeek() {
+  const now = DateTime.now().setZone('America/Vancouver')
+  const weekStart = now.minus({ weeks: 1 }).startOf('week')
+  const weekEnd = now.minus({ weeks: 1 }).endOf('week')
+
+  return {
+    start: luxonToCalendarDate(weekStart),
+    end: luxonToCalendarDate(weekEnd)
+  }
+}
+
+function getLastMonth() {
+  const now = DateTime.now().setZone('America/Vancouver')
+  const monthStart = now.minus({ months: 1 }).startOf('month')
+  const monthEnd = now.minus({ months: 1 }).endOf('month')
+
+  return {
+    start: luxonToCalendarDate(monthStart),
+    end: luxonToCalendarDate(monthEnd)
+  }
+}
+
+function resetRange() {
+  return { start: undefined, end: undefined }
+}
+
+// take local CalendarDate, convert to iso date and set v-model, close popover
+function applyDateRange() {
+  if (!localModel.value.start || !localModel.value.end) {
+    return
+  }
+  const start = localModel.value.start.toString()
+  const end = localModel.value.end.toString()
+  model.value = { start, end }
+  open.value = false
+}
+
+function cancel() {
+  localModel.value = resetRange()
+  open.value = false
+}
+
+// find the filter code in the config array or return custom range
+function getFilterCodeForRange(range?: { start?: CalendarDate, end?: CalendarDate }): string | undefined {
+  if (!range?.start || !range?.end) {
+    return DATEFILTER_CODES.CUSTOMRANGE
+  }
+  const startStr = range.start.toString()
+  const endStr = range.end.toString()
+
+  const matchedRange = config.find((c) => {
+    const r = c.getRange()
+    if (r.start && r.end) { // custom range sets to undefined
+      return startStr === r.start.toString() && endStr === r.end.toString()
+    }
+    return false
+  })
+  return matchedRange?.code || DATEFILTER_CODES.CUSTOMRANGE
+}
+
+// sync v-model and localModel state
```

Comment:
why not just name your function what your comment is? thus negating the need for it?